ddtrace/internal/writer.py

```diff
@@ -20,6 +22,63 @@
 LOG_ERR_INTERVAL = 60
 
 
+def _apply_filters(filters, traces):
+    """
+    Here we make each trace go through the filters configured in the
+    tracer. There is no need for a lock since the traces are owned by the
+    AgentWriter at that point.
+    """
+    if filters is not None:
+        filtered_traces = []
+        for trace in traces:
+            for filtr in filters:
+                trace = filtr.process_trace(trace)
+                if trace is None:
+                    break
+            if trace is not None:
+                filtered_traces.append(trace)
+        return filtered_traces
+    return traces
+
+
+class LogWriter:
+    def __init__(self, out=sys.stdout, filters=None, sampler=None, priority_sampler=None):
+        self._filters = filters
+        self._sampler = sampler
+        self._priority_sampler = priority_sampler
+        self.encoder = JSONEncoderV2()
+        self.out = out
+
+    def recreate(self):
+        """ Create a new instance of :class:`LogWriter` using the same settings from this instance
+
+        :rtype: :class:`LogWriter`
+        :returns: A new :class:`LogWriter` instance
+        """
+        writer = self.__class__(
+            out=self.out, filters=self._filters, sampler=self._sampler, priority_sampler=self._priority_sampler
+        )
+        return writer
+
+    def write(self, spans=None, services=None):
+        # We immediately flush all spans
+        if not spans:
+            return
+
+        # Before logging the traces, make them go through the
+        # filters
+        try:
+            traces = _apply_filters(self._filters, [spans])
+        except Exception:
+            log.error("error while filtering traces", exc_info=True)
+            return
+        if len(traces) == 0:
+            return
+        encoded = self.encoder.encode_traces(traces)
+        self.out.write("%s\n" % encoded)
```

Comment:
I wonder if it would be faster to do `encoded + "\n"`...

(I literally have no idea, just a musing)